plugins/net_plugin/net_plugin.cpp

```diff
@@ -2139,6 +2118,45 @@ namespace eosio {
       }
    }
 
+   bool net_plugin_impl::process_next_message(const connection_ptr& conn, uint32_t message_length) {
+      try {
+         // if next message is a block we already have, exit early
+         auto peek_ds = conn->pending_message_buffer.create_peek_datastream();
+         unsigned_int which{};
+         fc::raw::unpack( peek_ds, which );
```

Comment:
I think it would be good to have an assert on the only 2 valid types right now, since below code treats the 2 types as a bool.

Reply:
The valid types are 0-8. Could assert that range, however it is already asserted via `fc::raw::unpack` of `static_variant` when `set_which` of `static_variant` asserts range. That happens on line 2143 where `net_message` (which is a `static_variant`) is unpacked.